### PR TITLE
refactor: 💡 update to allow state change & update edit action

### PR DESCRIPTION
**Pull request by lisbet-alvarez**
✅ Closes: ICU-10078 & 10079

:tickets: [Jira ticket](https://hashicorp.atlassian.net/browse/ICU-10078)
:tickets: [Jira ticket](https://hashicorp.atlassian.net/browse/ICU-10079)

## Description
Allows state change for ldap auth-method.
Verifies that cancelling changes rolls back attributes.
Verifies that deleting ldap auth method is successful.
Verifies that editing an ldap auth method is successful.
Acceptance tests for delete and update.

:technologist: [Admin preview](https://boundary-ui-git-icu-10078-enable-edit-mode-hashicorp.vercel.app/scopes/global/auth-methods/am_ec8st2tp51)

### CODE REVIEW

File: addons/api/addon/serializers/auth-method.js [hunk 1]
```diff
@@ -51,6 +57,31 @@ export default class AuthMethodSerializer extends ApplicationSerializer {
...
 
+  /**
+   * If `adapterOptions.state` is set, the serialization should
+   * include **only state** and version.  Normally, this is not serialized.
+   * @param {Snapshot} snapshot
+   * @return {object}
+   */
+  serializeLDAP(snapshot) {
+    let serialized = super.serialize(...arguments);
+    // Deleting these attributes is a temporary fix.
```
Comment (lisbet-alvarez) [inline addons/api/addon/serializers/auth-method.js:68]: Changes in this file are temporary. The reason being that sending these fields to the backend prevents a user from being able to edit the form. There is also a bug related to this, visit this [slack thread](https://hashicorp.slack.com/archives/C014AHXUL3G/p1690405498990289) for more info.

File: addons/api/addon/models/auth-method.js [hunk 2]
```diff
@@ -118,9 +117,11 @@ export default class AuthMethodModel extends GeneratedAuthMethodModel {
    */
   changeState(state, options = { adapterOptions: {} }) {
     const defaultAdapterOptions = {
-      method: 'change-state',
       state,
     };
+    if (this.isOIDC) {
```
Comment (DhariniJeeva) [inline addons/api/addon/models/auth-method.js:122]: why is this only for OIDC?

File: addons/api/addon/serializers/auth-method.js (hunk 1 above)
Comment (DhariniJeeva) [reply addons/api/addon/serializers/auth-method.js:68]: I'm ok with a temp fix for now so it does not break the UI, could we add a ticket to track this? The ticket can be linked to the BE ticket so we can clean this up once the fix from BE is in.

File: addons/api/addon/models/auth-method.js (hunk 2 above)
Comment (lisbet-alvarez) [reply addons/api/addon/models/auth-method.js:122]: great question, only oidc has a 'change-state' endpoint. For ldap auth-methods, specifying a method causes the API to return an error. 
image

**Review by joshuaogle — APPROVED** [review]
Looks nice!

File: addons/api/addon/models/auth-method.js (hunk 2 above)
Comment (DhariniJeeva) [reply addons/api/addon/models/auth-method.js:122]: So we can't change the LDAP state after creation?

Comment (lisbet-alvarez) [reply addons/api/addon/models/auth-method.js:122]: Ldap state can be updated, its just that specifying the 'change-state' method for ldap will cause an error. The `serializeWithState` method in the serializer takes care of sending a payload with just the state.

**Review by dvonanderson — APPROVED** [review]
Awesome work Lis

**Pull request merged**